File: frontend/src/api/suggest.ts
import { apiFetch } from './client'

export interface SuggestionOut {
  id: string
  title: string
  excerpt: string
  source: string
  source_url: string
  type: string
  relevance: number
  tags: string[]
  provider: string
}

export interface SuggestRequest {
  query?: string
  offset?: number
  limit?: number
}

export const fetchSuggestions = (topicSlugOrId: string, body: SuggestRequest) =>
  apiFetch<SuggestionOut[]>(`/topics/${encodeURIComponent(topicSlugOrId)}/suggest`, {
    method: 'POST',
    body: JSON.stringify({
      query: body.query ?? '',
      offset: body.offset ?? 0,
      limit: body.limit ?? 5,
    }),
  })

File: backend/app/schemas/suggest.py
from pydantic import BaseModel, Field


class SuggestRequest(BaseModel):
    query: str = ""               # optional refine query from the sidebar input
    offset: int = 0
    limit: int = Field(5, ge=1, le=20)


class SuggestionOut(BaseModel):
    id: str
    title: str
    excerpt: str
    source: str
    source_url: str
    type: str = "Article"
    relevance: float = 0.5
    tags: list[str] = []
    provider: str = ""
